arbnode/transaction_streamer.go

```diff
@@ -1834,21 +1843,78 @@ func (s *TransactionStreamer) Start(ctxIn context.Context) error {
 }
 
 const ESPRESSO_TRANSACTION_SIZE_LIMIT int = 10 * 1024
+const MAX_ATTESTATION_QUOTE_SIZE int = 4 * 1024
 
-func buildHotShotPayload(msgs *[]arbostypes.L1IncomingMessage) (espressoTypes.Bytes, int) {
+func (t *TransactionStreamer) buildHotShotPayload(msgs *[]arbostypes.MessageWithMetadata, submittedTxnPos []arbutil.MessageIndex) (espressoTypes.Bytes, int) {
```

Comment:
I would like to keep the function pure. This will be easier for us to test this function though we don't have a unit test for this.

Reply:
We need to call getAttestationQuote which requires the `QuoteFile` and `AttestationFile` which are present in the config so this cant be a pure function